src/multirust-cli/proxy_mode.rs

```diff
@@ -0,0 +1,28 @@
+use common::{run_inner, set_globals};
+use multirust::{Cfg, Result, Error};
+use multirust_utils::utils;
+use std::env;
+use std::path::PathBuf;
+
+pub fn main() -> Result<()> {
+    try!(::self_update::cleanup_self_updater());
+
+    let arg0 = env::args().next().map(|a| PathBuf::from(a));
```

Comment:
Perhaps this could thread through the OsStr to down below? Probably doesn't matter here anyway but may matter someday and is arguably "best practice"